compiler+runtime/src/cpp/jank/runtime/core/seq.cpp

```diff
@@ -790,6 +790,10 @@ namespace jank::runtime
         }
         else if constexpr(behavior::seqable<T>)
         {
+          if constexpr(!behavior::sequential<T> && !std::is_same_v<T, obj::persistent_string>)
```

Comment:
Since `persistent_string` implements `associatively_readable` I'm going to go out on a limb and say it should also implement `indexable` (even though it doesn't in Clojure/script).

I think then we can make the condition here `(behavior::seqable<T> && behavior::sequential<T>)` and then remove the extra throws.

Reply:
Seems reasonable to me.

Reply:
Yeah, as you noted, I took the approach that felt most similar to Clojure, but it does make a lot of sense that `persistent_string` would have `nth` implemented. I'll go that way.